frontend/src/Pages/GithubOauth.js

```diff
@@ -0,0 +1,37 @@
+import React, { useEffect } from 'react'
+import { Navigate, useNavigate, useSearchParams } from 'react-router-dom';
+import axios from '../HelperFunctions/customAxios';
+
+export default function GithubOauth() {
+    const navigate = useNavigate();
+    const [searchParams, setSearchParams] = useSearchParams();
+    const loginCode = searchParams.get('code'), stateValue = searchParams.get('state');
+
+    const localStorageState = localStorage.getItem('state');
+    console.log(localStorageState, stateValue);
+
+    if(localStorageState !== stateValue){
+        console.error("State values do not match");
+        return (
+            <Navigate to="/" />
+        );
+    }
+
+    useEffect( () => {
+        async function sendCode(){
+            await axios.post('api/github/login', {
+                "code": loginCode
+            })
+            .catch( (err) => {
+                console.error(err);
+            });
+        }
+        const resp = sendCode();
+        resp.then( (resp) => {
+            navigate('/');
+        });
+    });
+    return (
```

Comment:
Is this a valid state? Should we have an error code here or a message like "you are being redirected"?

Reply:
I've added a text message in HTML for it 👍

https://github.com/MatrixDoctors/matrix-cerberus/blob/398e3af73cfb6f7867b9553e7d9de7d8d8c86300/frontend/src/Pages/GithubOauth.js#L32-L35